cypress/integration/buy-shirt.spec.ts

```diff
@@ -1,19 +1,26 @@
+import {MenuContentPage, ProductListPage, ShoppingCartPage,
+  LoginPage, AddressPage, ShippingPage, PaymentPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const producrListPage = new ProductListPage();
+const shoppingCartPage = new ShoppingCartPage();
+const loginPage = new LoginPage();
+const addressPage = new AddressPage();
+const shippingPage = new ShippingPage();
+const paymentPage = new PaymentPage();
+
 describe("Buy a t-shirt", () => {
   it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click();
-    cy.get("[style*=\"display: block;\"] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-    cy.get("#SubmitLogin > span").click();
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get("#cgv").click();
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get(".bankwire").click();
-    cy.get("#cart_navigation > .button > span").click();
-    cy.get("#center_column > div > p > strong")
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+    producrListPage.addTShirtToCart();
+    shoppingCartPage.processToCheckOut();
+    loginPage.login();
```

Comment:
the login method should receive the email and password as parameters, in that way the page object is not attached to a specific user

Reply:
Done!